test/micro/org/openjdk/bench/jdk/incubator/vector/MatrixMultiplicationBenchmark.java

```diff
@@ -0,0 +1,185 @@
+                    for (int j = jj; j < jj + blocksize; ++j) {
+                        float sum = result[i * n + j];
+                        for (int k = kk; k < kk + blocksize; ++k) {
+                            sum += a[i * n + k] * b[k * n + j];
+                        }
+                        result[i * n + j] = sum;
+                    }
+                }
+            }
+        }
+        return result;
+    }
+
+    private float[] simpleFMA(float[] a, float[] b, float[] result, int n) {
+        int in = 0;
+        for (int i = 0; i < n; ++i) {
+            int kn = 0;
+            for (int k = 0; k < n; ++k) {
+                float aik = a[in + k];
+                for (int j = 0; j < n; ++j) {
+                    result[in + j] = Math.fma(aik, b[kn + j], result[in + j]);
+                }
+                kn += n;
+            }
+            in += n;
+        }
+        return result;
+    }
+
+    private float[] simpleVector(float[] a, float[] b, float[] result, int n) {
+        final int upperBound = SPECIES.loopBound(n);
+
+        int in = 0;
+        for (int i = 0; i < n; i++) {
+            int kn = 0;
+            for (int k = 0; k < n; k++) {
+                float aik = a[in + k];
+                FloatVector vaik = FloatVector.broadcast(SPECIES, aik);
+
+                for (int j = 0; j < upperBound; j += SPECIES.length()) {
+                    // FloatVector va, vb, vc
+                    var vb = FloatVector.fromArray(SPECIES, b, kn + j);
+                    var vResult = FloatVector.fromArray(SPECIES, result, in + j);
+                    vResult = vaik.fma(vb, vResult);
+                    vResult.intoArray(result, in + j);
+                }
+                kn += n;
+            }
+            in += n;
+        }
+        return result;
+    }
+
+    private float[] blockedVector(float[] a, float[] b, float[] result, int n) {
+        int blockWidth = n >= 256 ? 512 : 256;
+        int blockHeight = n >= 512 ? 8 : n >= 256 ? 16 : 32;
+
+        for (int rowOffset = 0; rowOffset < n; rowOffset += blockHeight) {
+            for (int columnOffset = 0; columnOffset < n; columnOffset += blockWidth) {
+                for (int i = 0; i < n; i++) {
+                    for (int j = columnOffset; j < columnOffset + blockWidth && j < n; j += SPECIES.length()) {
+                        var sum = FloatVector.fromArray(SPECIES, result, i * n + j);
+                        for (int k = rowOffset; k < rowOffset + blockHeight && k < n; k++) {
+                            var multiplier = FloatVector.broadcast(SPECIES, a[i * n + k]);
+                            sum = multiplier.fma(FloatVector.fromArray(SPECIES, b, k * n + j), sum);
+                        }
+                        sum.intoArray(result, i * n + j);
+                    }
+                }
+            }
+        }
+        return result;
+    }
+
+    private static float[] newFloatRowMatrix(int size) {
+        Random rand = new Random();
```

Comment:
Instead we can use the new random generator API e.g,:
```
var rand = RandomGenerator.getDefault();
```
This is not super important, but its good practice to try and use more preferred APIs.